core/src/main/java/hudson/PluginManager.java

```diff
@@ -1643,26 +1649,73 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
     @Restricted(NoExternalUse.class)
     @RequirePOST public HttpResponse doCheckUpdatesServer() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
-        try {
-            for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
-                FormValidation v = site.updateDirectlyNow(DownloadService.signatureCheck);
-                if (v.kind != FormValidation.Kind.OK) {
-                    // TODO crude but enough for now
-                    return v;
-                }
+
+        int sleep = SystemProperties.getInteger(PluginManager.class.getName() + ".sleepTimeMilis", 1000);
+        int retries = SystemProperties.getInteger(PluginManager.class.getName() + ".retries", 3);
+
+        retries = retries > 0 ? retries : 3;
+        sleep = sleep > 0 ? sleep : 1000;
+
+        FormValidation response = null;
+        int i = 0;
+        boolean success = false;
+        while(i < retries && !success) {
+            i++;
+            try {
+                response = checkUpdatesServer();
+                lastErrorCheckUpdateCenters = response.getMessage(); //may be null (OK)
+                success = (response.kind == FormValidation.Kind.OK);
+            } catch (Exception e) {
+                // No matter what went on, don't show angry jenkins, just a message and see log for more info
+                lastErrorCheckUpdateCenters = e.getLocalizedMessage();
             }
-            for (DownloadService.Downloadable d : DownloadService.Downloadable.all()) {
-                FormValidation v = d.updateNow();
-                if (v.kind != FormValidation.Kind.OK) {
-                    return v;
+
+            if (!success) {
```

Comment:
not sure the role of `success` is required? Couldn't you move all this above in the try block, and finish with a `break;` to just go out of the loop in case of success?